controllers/snapshot/snapshot_adapter.go

```diff
@@ -307,6 +308,27 @@ func (a *Adapter) EnsureCreationOfEphemeralEnvironments() (controller.OperationR
 					"integrationTestScenario.Name", integrationTestScenario.Name)
 				return controller.StopProcessing()
 			}
+			if clienterrors.IsInvalid(err) {
+
+				if !gitops.IsSnapshotMarkedAsFailed(a.snapshot) {
+					var (
+						errKind   string = "<unknown>"
+						errName   string = "<unknown>"
+						statusErr *clienterrors.StatusError
+					)
+					if ok := errors.As(err, &statusErr); ok {
+						errKind = statusErr.Status().Details.Kind
+						errName = statusErr.Status().Details.Name
+					}
```

Comment:
nit/suggestion: this way of extracting basic info from errors seems like it could be useful in other places too, so it could be a good idea to extract it into a generic function.

Reply:
I'm not sure if this can be a generic function, it would work only for StatusError, but for better readability, we can introduce `GetDetailsFromStatusError` to return `Details`

Reply:
I refactored out that bit to make things a bit more readable, but not sure how I would implement a generic here. Im interested in your thoughts @dirgim & @MartinBasti

Reply:
Yes, agreed.